pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -375,70 +409,83 @@ func (r *ReconcileUnifiedPushServer) Reconcile(request reconcile.Request) (recon
-		return reconcile.Result{}, nil
 	} else if err != nil {
 		return reconcile.Result{}, err
-	} else {
-		desiredUnifiedPushImage := unifiedpush.image()
-		desiredProxyImage := proxy.image()
+	}
+	//#endregion
 
-		unifiedPushContainerSpec := findContainerSpec(foundUnifiedpushDeployment, UPS_CONTAINER_NAME)
-		if unifiedPushContainerSpec == nil {
-			reqLogger.Info("Skipping image reconcile: Unable to find container spec in deployment", "Deployment.Namespace", foundUnifiedpushDeployment.Namespace, "Deployment.Name", foundUnifiedpushDeployment.Name, "ContainerSpec", UPS_CONTAINER_NAME)
-		} else if unifiedPushContainerSpec.Image != desiredUnifiedPushImage {
-			reqLogger.Info("Container spec in deployment is using a different image. Going to update it now.", "Deployment.Namespace", foundUnifiedpushDeployment.Namespace, "Deployment.Name", foundUnifiedpushDeployment.Name, "ContainerSpec", UPS_CONTAINER_NAME, "ExistingImage", unifiedPushContainerSpec.Image, "DesiredImage", desiredUnifiedPushImage)
+	//#region UPS ImageStream
+	unifiedPushImageStream, err := newUnifiedPushImageStream(instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
 
-			// update
-			updateContainerSpecImage(foundUnifiedpushDeployment, UPS_CONTAINER_NAME, desiredUnifiedPushImage)
+	// Set UnifiedPushServer instance as the owner and controller
+	if err := controllerutil.SetControllerReference(instance, unifiedPushImageStream, r.scheme); err != nil {
+		return reconcile.Result{}, err
+	}
 
-			// enqueue
-			err = r.client.Update(context.TODO(), foundUnifiedpushDeployment)
-			if err != nil {
-				reqLogger.Error(err, "Failed to update Deployment", "Deployment.Namespace", foundUnifiedpushDeployment.Namespace, "Deployment.Name", foundUnifiedpushDeployment.Name)
-				return reconcile.Result{}, err
-			}
-			return reconcile.Result{Requeue: true}, nil
+	// Check if this ImageStream already exists
+	foundUnifiedPushImageStream := &imagev1.ImageStream{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: unifiedPushImageStream.Name, Namespace: unifiedPushImageStream.Namespace}, foundUnifiedPushImageStream)
+	if err != nil && errors.IsNotFound(err) {
+		reqLogger.Info("Creating a new ImageStream", "ImageStream.Namespace", unifiedPushImageStream.Namespace, "ImageStream.Name", unifiedPushImageStream.Name)
+		err = r.client.Create(context.TODO(), unifiedPushImageStream)
+		if err != nil {
+			return reconcile.Result{}, err
 		}
+	} else if err != nil {
+		return reconcile.Result{}, err
+	}
+	//#endregion
 
-		proxyContainerSpec := findContainerSpec(foundUnifiedpushDeployment, OAUTH_PROXY_CONTAINER_NAME)
-		if proxyContainerSpec == nil {
-			reqLogger.Info("Skipping image reconcile: Unable to find container spec in deployment", "Deployment.Namespace", foundUnifiedpushDeployment.Namespace, "Deployment.Name", foundUnifiedpushDeployment.Name, "ContainerSpec", OAUTH_PROXY_CONTAINER_NAME)
-		} else if proxyContainerSpec.Image != desiredProxyImage {
```

Comment:
Image reconciliation done with https://issues.jboss.org/browse/AEROGEAR-9012 doesn't make sense anymore with ImageStreams.
Got rid of this block.